### PR TITLE
Metrics

### PR DESCRIPTION
**Pull request by chuckwired**
- [x] space in models for metrics
- [x] withMetric method to attach received measurement to that set's exercises
- [x] iOS marshalling

### CODE REVIEW

File: server/exercise/src/main/scala/com/eigengo/lift/exercise/UserExercisesSessions.scala [hunk 1]
```diff
@@ -30,6 +31,11 @@ object UserExercisesSessions {
       if (kie.isEmpty) 0.5 else kie.sum / kie.size
     }
 
+    def withMetric(metric: Metric): ExerciseSet = {
+      val newList = exercises.map(e ⇒ Exercise(e.name, e.intensity, Some(metric)))
+      ExerciseSet(newList)
```
Comment (janm399) [inline server/exercise/src/main/scala/com/eigengo/lift/exercise/UserExercisesSessions.scala]: Great; or to stay with the style below, use 
```scala
val newExercises = exercises.map(e => Exercise(...))
copy(exercises = newExercises)
```

Comment (chuckwired) [inline server/exercise/src/main/scala/com/eigengo/lift/exercise/UserExercisesSessions.scala]: Ah okay, has been pushed

File: ios/Lift/LiftServer/ExerciseModel+Marshalling.swift [hunk 2]
```diff
@@ -31,13 +31,19 @@ extension Exercise.SessionProps {
 extension Exercise.Exercise {
     
     static func unmarshal(json: JSON) -> Exercise.Exercise {
-        return Exercise.Exercise(name: json["name"].stringValue, intensity: json["intensity"].double)
+        return Exercise.Exercise(name: json["name"].stringValue, intensity: json["intensity"].double, metric: Exercise.Metric(value: json["metric"]["value"].doubleValue, metricUnit: json["metric"]["metricUnit"].stringValue))
```
Comment (janm399) [inline ios/Lift/LiftServer/ExerciseModel+Marshalling.swift]: It'd be better if you added ``unmarshal(json: JSON) -> Exercise.MetricUnit`` to its own extension and use it here.

Comment (chuckwired) [inline ios/Lift/LiftServer/ExerciseModel+Marshalling.swift]: Added! and an Exercise.Metric.marshal()

File: ios/Lift/LiftServer/ExerciseModel+Marshalling.swift [hunk 3]
```diff
@@ -31,17 +31,32 @@ extension Exercise.SessionProps {
...
 
+extension Exercise.Metric {
+
+    static func unmarshal(json: JSON) -> Exercise.Metric {
+        return Exercise.Metric(value: json["value"].doubleValue, metricUnit: json["metricUnit"].stringValue)
+    }
+    
+    func marshal() -> [String : AnyObject] {
+        var metricParams: [String : AnyObject] = [:]
+        metricParams["value"] = value ?? 0.0
```
Comment (janm399) [inline ios/Lift/LiftServer/ExerciseModel+Marshalling.swift]: You don't need the optionals. The ``value`` and ``metricUnit`` are always defined here.

File: ios/Lift/LiftServer/ExerciseModel+Marshalling.swift [hunk 4]
```diff
@@ -31,17 +31,32 @@ extension Exercise.SessionProps {
 extension Exercise.Exercise {
     
     static func unmarshal(json: JSON) -> Exercise.Exercise {
-        return Exercise.Exercise(name: json["name"].stringValue, intensity: json["intensity"].double)
+        return Exercise.Exercise(name: json["name"].stringValue, intensity: json["intensity"].double, metric: Exercise.Metric.unmarshal(json["metric"]))
```
Comment (janm399) [inline ios/Lift/LiftServer/ExerciseModel+Marshalling.swift]: :+1:

**Pull request merged**